bento/src/main/java/com/yelp/android/bento/components/ListComponent.java

```diff
@@ -85,18 +85,31 @@ public void appendData(@NonNull List<T> data) {
         notifyItemRangeInserted(oldSize, sizeChange);
     }
 
+    /**
+     * Inserts the specified data at the specified index in the component data.
+     *
+     * @param index the index to insert the given data
+     * @param data the data to be inserted
+     */
+    public void insertData(int index, @NonNull T data) {
+        mData.add(index, data);
+        notifyItemRangeInserted(index, index + 1);
```

Comment:
This needs to take dividers into account. Pretty sure the first argument is fine, but the second needs to be either +1 or +2 if there is a divider.

Reply:
Got it! Looks like as long as the itemCount is right, the index doesn't seem to matter. Might look into that another time